pkg/cert/v1beta1/generate/generate.go

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright 2021 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generate
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"github.com/kubeflow/katib/pkg/cert/v1beta1/consts"
+	"github.com/spf13/cobra"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+	"math/big"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+)
+
+type generateOptions struct {
+	namespace         string
+	fullServiceDomain string
+}
+
+func NewGenerateCmd(kubeClient client.Client) *cobra.Command {
+	o := &generateOptions{}
+	cmd := &cobra.Command{
+		Use:          "generate",
+		Short:        "generate server cert for webhook",
+		Long:         "generate server cert for webhook",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.run(context.TODO(), kubeClient); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	f := cmd.Flags()
+	f.StringVarP(&o.namespace, "namespace", "n", "kubeflow", "set namespace")
+	return cmd
+}
+
+func (o *generateOptions) run(ctx context.Context, kubeClient client.Client) error {
+	o.fullServiceDomain = strings.Join([]string{consts.Service, o.namespace, "svc"}, ".")
+
+	caKeyPair := &certificates{}
```

Comment:
Do we need to send empty `&certificates{}` to `createCACert` function ?
Is there any use-case when we should populate `certificates{}` before executing `createCACert` ?
If not, maybe it makes sense to just return `*certificates` from the `createCACert` function ?

WDYT @tenzen-y ?

Reply:
> Is there any use-case when we should populate certificates{} before executing createCACert ?
If not, maybe it makes sense to just return *certificates from the createCACert function ?

Currently, there is no such use-case. So, your suggestion makes sense.

Reply:
I fixed it.